windows/RNLocalize/RNLocalizeModule.cpp

```diff
@@ -0,0 +1,212 @@
+{
+  return !winrt::to_string(winrt::Windows::Globalization::Calendar().GetClock()).compare("24HourClock");
+}
+
+std::vector<std::string> RNLocalizeModule::getCurrencies()
+{
+  std::vector<std::string> currencies;
+  for (winrt::hstring currency : winrt::Windows::Globalization::GeographicRegion().CurrenciesInUse())
+  {
+    currencies.push_back(winrt::to_string(currency));
+  }
+  return currencies;
+}
+
+std::string RNLocalizeModule::getScriptCode(winrt::hstring locale)
+{
+  return winrt::to_string(winrt::Windows::Globalization::Language(locale).Script());
+}
+
+std::string RNLocalizeModule::getCalendar()
+{
+  std::string calendar = to_string(winrt::Windows::Globalization::Calendar().GetCalendarSystem());
+
+  // Window's Calendar system can return any of these values:
+  // "EastAsianLunisolarCalendar", "GregorianCalendar", "HebrewCalendar, "HijriCalendar",
+  // "JapaneseCalendar", "JulianCalendar", "KoreanCalendar", "PersianCalendar",
+  // "TaiwanCalendar", "ThaiBuddhistCalendar", "UmAlQuraCalendar"
+  // In all cases they end in "Calendar" and RNL doesn't expect this, so we must truncate it.
+  calendar = calendar.substr(0, calendar.size() - 8);
+
+  // Special edge case: RNL expects "buddhist" not "thaibuddhist"
+  // see https://github.com/zoontek/react-native-localize/blob/master/src/types.ts
+  if (calendar == "ThaiBuddhist")
+  {
+    return "buddhist";
+  }
+
+  std::transform(calendar.begin(), calendar.end(), calendar.begin(),
+                 [](char c) {
+                   return static_cast<char>(std::tolower(c));
+                 });
+
+  return calendar;
+}
+
+std::string RNLocalizeModule::getTimeZone()
+{
+  // Use GetTimeZoneInformation to get the hour bias from GMT and then create a formatted string.
+  TIME_ZONE_INFORMATION TimeZoneInfo;
+  GetTimeZoneInformation(&TimeZoneInfo);
+
+  int hourBias = -TimeZoneInfo.Bias / 60;
+  char timeZoneBuffer[10];
+  snprintf(
+      timeZoneBuffer,
+      sizeof(timeZoneBuffer),
+      "GMT%s%02d:00",
+      hourBias < 0 ? "-" : "+",
+      hourBias < 0 ? -hourBias : hourBias);
+  std::string timeZoneString(timeZoneBuffer);
+
+  return timeZoneString;
+}
+
+NumberFormatSettings RNLocalizeModule::getNumberFormatSettings(std::string locale)
+{
+  NumberFormatSettings numberFormatSettings;
+
+  // Use GetNumberFormatEx to get number formatting based on the passed locale.
+  std::string num_s("1000.00");
+  std::wstring num_w = std::wstring(num_s.begin(), num_s.end());
+
+  const wchar_t *locale_l = std::wstring(locale.begin(), locale.end()).c_str();
+  const int numberFormatBufferLength = 9;
+  wchar_t numberFormatBuffer[numberFormatBufferLength];
+  GetNumberFormatEx(locale_l, 0, num_w.c_str(), nullptr, numberFormatBuffer, numberFormatBufferLength);
+  std::string formattedString = winrt::to_string(numberFormatBuffer);
+
+  numberFormatSettings.decimalSeparator = formattedString.substr(5, 1);
+  numberFormatSettings.groupingSeparator = formattedString.substr(1, 1);
+
```

Comment:
not sure if there is a JS api for this here, but different locales have different rules for _where_ to put the grouping separator
for example in India you'd write "1,000" and "10,000", but "1,00,000" for 100000

Reply:
There isn't an API in this library as far as I can tell to indicate how that should be handled.

Might need to be hardcoded per language if that ever gets added.

And most of the more powerful friendly looking windows api's aren't actually available to us in this context .

So we had much more limited options for this :(